app/src/main/res/layout/activity_login.xml

```diff
@@ -42,12 +42,60 @@
                     android:scaleType="fitXY" />
 
                 <android.support.design.widget.TextInputLayout
-                    android:id="@+id/username_layout"
+                    android:id="@+id/login_process_layout"
                     android:layout_width="wrap_content"
                     android:layout_height="wrap_content"
                     android:layout_below="@id/icon_iv"
                     android:layout_centerInParent="true"
                     android:textColor="@color/colorPrimaryDark"
+                    android:visibility="@{VIEW.VISIBLE}">
+
+                    <android.support.v7.widget.AppCompatEditText
+                        android:id="@+id/generic_login_error_tv"
+                        android:layout_width="300dp"
+                        android:layout_height="0dp"
+                        android:inputType="text" />
+                </android.support.design.widget.TextInputLayout>
+
+                <android.support.design.widget.TextInputLayout
+                    android:id="@+id/client_secret_layout"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_below="@id/login_process_layout"
+                    android:layout_centerInParent="true"
+                    android:textColor="@color/colorPrimaryDark"
+                    android:visibility="@{VIEW.VISIBLE}">
```

Comment:
I'm only using them to get a common interface to setErrorEnabled and setError, made them invisibly by having them be zero height. If you have better suggestions, or better styling recommendations, let me know.

Primarily i think using toasts in that area for reporting errors is not great as it precludes users from making screenshots if they're not fast enough.